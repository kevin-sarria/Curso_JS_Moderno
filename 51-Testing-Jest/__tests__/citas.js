import Citas from '../js/classes/Citas.js';

describe('Probar la clase de citas', () => {

    const citas = new Citas();

    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'asdfs',
            propietario: 'asfd',
            telefono: '1234567890',
            fecha: '10-10-2010',
            hora:'10:30',
            sintomas: 'asvdc'
        }

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();

    });

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'asdfs',
            propietario: 'asfd',
            telefono: '1234567890',
            fecha: '10-10-2010',
            hora:'10:30',
            sintomas: 'asvdc'
        }

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();

    });

    test('Eliminar Cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();

    });

});


