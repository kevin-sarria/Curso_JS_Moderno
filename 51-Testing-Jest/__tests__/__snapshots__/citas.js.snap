// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-10-2010",
      "hora": "10:30",
      "id": 1656371664234,
      "mascota": "asdfs",
      "propietario": "asfd",
      "sintomas": "asvdc",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-10-2010",
      "hora": "10:30",
      "id": 1656371664234,
      "mascota": "asdfs",
      "propietario": "asfd",
      "sintomas": "asvdc",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;
